cli/src/trusted_commands.rs

```diff
@@ -540,3 +622,134 @@ fn unshield_funds(
+		967295,        // gas limit
+		U256::from(1), // max_fee_per_gas !>= min_gas_price defined in runtime
+		None,
+		None,
+		Vec::new(),
+	)
+	.sign(&from.into(), nonce, &mrenclave, &shard)
+	.into_trusted_operation(trusted_args.direct);
+
+	let _ = perform_operation(cli, trusted_args, &top);
+
+	let execution_address = evm_create_address(sender_evm_acc, evm_account_nonce);
+	info!("trusted call evm_create executed");
+	println!("Hopefully created the smart contract with address {:?}", execution_address);
+}
+
+#[cfg(feature = "evm")]
+fn evm_read_storage(
+	cli: &Cli,
+	trusted_args: &TrustedArgs,
+	arg_from: &str,
+	execution_address_str: &str,
+) {
+	let sender = get_pair_from_str(trusted_args, arg_from);
+	let sender_acc: AccountId = sender.public().into();
+
+	println!("senders ss58 is {}", sender.public().to_ss58check());
+
+	let mut sender_evm_acc_slice: [u8; 20] = [0; 20];
+	sender_evm_acc_slice
+		.copy_from_slice((<[u8; 32]>::from(sender_acc.clone())).get(0..20).unwrap());
+	let sender_evm_acc: H160 = sender_evm_acc_slice.into();
+
+	println!("senders evm account is {}", sender_evm_acc);
+
+	let execution_address =
+		H160::from_slice(&Vec::from_hex(execution_address_str.to_string()).unwrap());
+
+	let top: TrustedOperation =
+		TrustedGetter::evm_account_storages(sender_acc, execution_address, H256::zero())
+			.sign(&KeyPair::Sr25519(sender))
+			.into();
+	let res = perform_operation(cli, trusted_args, &top);
+
+	debug!("received result for balance");
+	let val = if let Some(v) = res {
+		if let Ok(vd) = H256::decode(&mut v.as_slice()) {
+			vd
+		} else {
+			error!("could not decode value. {:x?}", v);
+			H256::zero()
+		}
+	} else {
+		error!("Nothing in state!");
+		H256::zero()
+	};
+
+	println!("{:?}", val);
+}
+
+#[cfg(feature = "evm")]
+fn evm_call(
+	cli: &Cli,
+	trusted_args: &TrustedArgs,
+	arg_from: &str,
+	execution_address_str: &str,
+	function: &str,
+) {
+	let sender = get_pair_from_str(trusted_args, arg_from);
+	let sender_acc: AccountId = sender.public().into();
+
+	println!("senders ss58 is {}", sender.public().to_ss58check());
+
+	let mut sender_evm_acc_slice: [u8; 20] = [0; 20];
+	sender_evm_acc_slice
+		.copy_from_slice((<[u8; 32]>::from(sender_acc.clone())).get(0..20).unwrap());
+	let sender_evm_acc: H160 = sender_evm_acc_slice.into();
+
+	println!("senders evm account is {}", sender_evm_acc);
+
+	let execution_address =
+		H160::from_slice(&Vec::from_hex(execution_address_str.to_string()).unwrap());
+
+	let function_hash = Vec::from_hex(function.to_string()).unwrap();
+
+	let (mrenclave, shard) = get_identifiers(trusted_args);
+	let nonce = get_layer_two_nonce!(sender, cli, trusted_args);
+
+	println!("calling smart contract function");
+
+	let inc_call = TrustedCall::evm_call(
+		sender_acc,
+		sender_evm_acc,
+		execution_address,
+		function_hash,
+		U256::from(0),
+		10_000_000,    // gas limit
+		U256::from(1), // max_fee_per_gas !>= min_gas_price defined in runtime
+		None,
+		Some(U256::from(nonce)),
```

Comment:
I think this parameter should be the value fetched with `get_layer_two_evm_nonce`

Reply:
Now the lower nonce is wrong (this one should be a polkadot nonce). We need both nonces (polkadot and ethereum)